config/initializers/session_store.rb

```diff
@@ -1,6 +1,22 @@
 # Be sure to restart your server when you modify this file.
 
-Foodsoft::Application.config.session_store :cookie_store, key: '_foodsoft_session'
+module ActionDispatch
+  module Session
+    class SlugCookieStore < CookieStore
+      alias_method :orig_set_cookie, :set_cookie
+
+      def set_cookie(env, session_id, cookie)
+        request = ActionDispatch::Request.new env
+        slug = request.original_fullpath.split('/', 3).second
+        return if slug.blank?
+        cookie[:path] = '/' + slug
```

Comment:
This would prevent installation running on a subdomain (I'm doing that currently). Would it be possible to get the base url from somewhere (I think it is already set in the mailer config, or else we _do_ have a request here, I guess)?

Reply:
hmm, good point. we are in a very early phase of the request handling, where we do not have all this information. so i'd suggest a new config variable. if that variable is not set, the old behaviour (without cookie path) remains

Reply:
Yes, we do have [`host` and such](https://github.com/foodcoops/foodsoft/blob/master/config/app_config.yml.SAMPLE#L134), could add `path` which defaults to `/`? (Right now I'm setting hostname to `hostname/path` which works for the mailer.)

Reply:
Yes, we do have [`host` and such](https://github.com/foodcoops/foodsoft/blob/master/config/app_config.yml.SAMPLE#L134), could add `path` which defaults to `/`? (Right now I'm setting hostname to `hostname/path` which works for the mailer.) Then `config/environments/production.rb` could include both hostname and path for hostname.